src/main/kotlin/eu/europa/ec/eudi/pidissuer/port/input/IssueCredential.kt

```diff
@@ -459,54 +460,63 @@ private fun ProofTo.toDomain(): UnvalidatedProof = when (type) {
-                        if (encryptionAlgorithm !in supported.parameters.algorithmsSupported) {
-                            raise(IllegalArgumentException("jwe encryption algorithm '${encryptionAlgorithm.name}' is not supported"))
-                        }
-                        if (encryptionMethod !in supported.parameters.methodsSupported) {
-                            raise(IllegalArgumentException("jwe encryption method '${encryptionMethod.name}' is not supported"))
-                        }
-                    }
+private fun RequestedResponseEncryption.ensureIsSupported(supported: CredentialResponseEncryption) {
+    try {
+        when (supported) {
+            is CredentialResponseEncryption.NotSupported -> {
+                if (this is RequestedResponseEncryption.Required) {
+                    // credential response encryption not supported by issuer but required by client
+                    throw IllegalArgumentException("credential response encryption is not supported")
                 }
+            }
 
-                is CredentialResponseEncryption.Required -> {
-                    if (this !is RequestedResponseEncryption.Required) {
-                        // credential response encryption required by issuer but not required by client
-                        raise(IllegalArgumentException("credential response encryption is required"))
-                    }
-
+            is CredentialResponseEncryption.Optional -> {
+                if (this is RequestedResponseEncryption.Required) {
+                    // credential response encryption supported by issuer and required by client
                     // ensure provided parameters are supported
                     if (encryptionAlgorithm !in supported.parameters.algorithmsSupported) {
-                        raise(IllegalArgumentException("jwe encryption algorithm '${encryptionAlgorithm.name}' is not supported"))
+                        throw IllegalArgumentException("jwe encryption algorithm '${encryptionAlgorithm.name}' is not supported")
                     }
                     if (encryptionMethod !in supported.parameters.methodsSupported) {
-                        raise(IllegalArgumentException("jwe encryption method '${encryptionMethod.name}' is not supported"))
+                        throw IllegalArgumentException("jwe encryption method '${encryptionMethod.name}' is not supported")
                     }
                 }
             }
-        }
 
-        RequestedResponseEncryption.Required(Json.encodeToString(key), algorithm, method)
-            .bind()
-            .also { it.ensureIsSupported() }
+            is CredentialResponseEncryption.Required -> {
+                if (this !is RequestedResponseEncryption.Required) {
+                    // credential response encryption required by issuer but not required by client
+                    throw IllegalArgumentException("credential response encryption is required")
```

Comment:
Please use error( .. ) instead of IllegalArgumentException

Further more, why use if ... throw and not the ensure()